tests/test_other.py

```diff
@@ -2417,6 +2417,18 @@ def test_embind(self):
       output = self.run_js('a.out.js')
       self.assertNotContained('FAIL', output)
 
+  def test_embind_finalization(self):
+    self.run_process(
+      [EMXX,
+       test_file('embind/test_finalization.cpp'),
+       '--post-js', test_file('embind/test_finalization.js'),
+       '--bind']
+    )
+    self.node_args += ['--expose-gc']
+    output = self.run_js('a.out.js', engine=config.NODE_JS)
+    self.assertContained('Constructed from C++ destructed', output)
+    self.assertContained('Constructed from JS destructed', output)
```

Comment:
We should probably also check that the pointer one is not destructed. Perhaps add IDs to each `Foo` instance (e.g. via constructor param), and then check that output contains `destructed #1` and `destructed #2` but not `destructed #3`.